lib/matplotlib/cm.py

```diff
@@ -88,7 +89,12 @@ def __getitem__(self, item):
         try:
             return self._cmaps[item].copy()
         except KeyError:
-            raise KeyError(f"{item!r} is not a known colormap name") from None
+            known = list(self._cmaps)
+            if len(best := difflib.get_close_matches(item, known, cutoff=.1)):
+                msg = f"Did you mean one of {best}?"
+            else:
+                msg = "No known keys are close"
```

Comment:
```suggestion
                msg = ""
```
Slight preference for not saying anyhing at all in this case. The added information is likely not too helpful "X is not a valid color name, *and it's not just a typo*." But if you want to say something, wording must be different. "key" and "close" are technical slang and implementation details that a typical user would know how to interpret.

Reply:
I don't think an exhaustive list would be readable. We have about 80 builtin colormaps (not including reversed). If you want to give access to a full list, better to use an indirection like

> For a full list use `list(mpl.colormaps)` or check https://matplotlib.org/stable/gallery/color/colormap_reference.html.